var/spack/repos/builtin/packages/heffte/package.py

```diff
@@ -165,6 +165,9 @@ def test(self):
                 "-Drocfft_DIR=" + join_path(self.spec["rocfft"].prefix, "lib", "cmake", "rocfft")
             )
 
+        # Provide the path to the MPI dependency so that the test can find an MPI launcher
+        options.append("-DCMAKE_PREFIX_PATH=" + self.spec["mpi"].prefix)
```

Comment:
Are you sure will work? CMake recommend you use `MPIEXEC_EXECUTABLE` and `MPI_CXX_COMPILER` for locating MPI.

https://cmake.org/cmake/help/latest/module/FindMPI.html

Reply:
Yes, this allows the cmake process that is building the tests to find the MPI install and to set the needed variables like MPIEXEC_EXECUTABLE. It is true that the user could in theory provide MPIEXEC_EXECUTABLE to cmake, but wouldn't make sense in this case because the value of MPIEXEC_EXECUTABLE is already determined and put into the cmake files that are installed with the MPI that was set up as a dependency of heffte. We set CMAKE_PREFIX_PATH so that the cmake build of the tests can find the previously installed MPI and get the variables that it needs from there.

Reply:
Normally, spack would do this automatically for packages that are set up as a runtime dependency, but the smoke tests are not currently treated correctly within spack (by getting the runtime dependencies automatically provided).

Reply:
As an example, here is another packages doing something very similar (but for a larger number of dependencies):
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/slate/package.py

Reply:
Ugh, more spack quirks with workarounds that will only break again the future.

Reply:
Yep, my understanding from the spack devs is that there will eventually be proper injection of dependencies for the smoke test environment, but for now it is all workarounds.  I'll be happy to remove all this once the proper functionality is in place!

Reply:
> Are you sure will work? CMake recommend you use `MPIEXEC_EXECUTABLE` and `MPI_CXX_COMPILER` for locating MPI.
> 
> https://cmake.org/cmake/help/latest/module/FindMPI.html

https://cmake.org/cmake/help/latest/module/FindMPI.html#variables-for-locating-mpi lists the second variable you can use as `MPI_HOME`, which I found works for me.